lib/dangermattic/plugins/check_tracks.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin for checking tracks-related changes and providing instructions in the PR if that's the case.
+  class CheckTracks < Plugin
+    DEFAULT_TRACKS_FILES = [
+      'AnalyticsTracker.kt',
+      'AnalyticsEvent.kt',
+      'LoginAnalyticsTracker.kt',
+      'WooAnalyticsStat.swift'
+    ].freeze
+
+    DEFAULT_TRACKS_USE_MATCHERS = [
+      /AnalyticsTracker\.track/
+    ].freeze
```

Comment:
(gathering replies from the [initial comment](https://github.com/Automattic/dangermattic/pull/15#pullrequestreview-1592553671) as it is kinda related to this one)

> If you haven't done it already, do you think it would be worth asking the Woo team whether they find value in this check?

I haven't. Since this is a very process-specific check, that's a good idea.

> Intuitively, this seems very useful

That's what I thought, but that was also an assumption based on what I saw in Peril. I'll write to them anyway.

One thing that I thought was that this check mostly builds on top of generic code and it could be a template for similar cases; so, even if this isn't applicable right now, this plugin could be easily repurposed to a generic case (as in: "warn/message/error" on files/patterns inside the diff).

> How do you feel about these default values?

I had mixed feelings about it, since it is very team-specific (Woo iOS/Android) and these plugin checks should be as general purpose as possible. So I thought it could be a repo/team specific check; alternatively, in this case, perhaps it would be even worth it moving this plugin code to Woo's Dangerfile.
And since this was running only on Woo, I'm not sure if it would be useful for other repos. Maybe for checking completely different things other than Tracks, with a different message?...

> In the context of a Danger plugin, I fell like these would be useful defaults only for WooCommerce

I think this is what is the key here: I am not sure yet how to deal with checks that are very team specific, or even to a couple of teams but not really general purpose.
In principle, my idea for these cases was to use shared specialised `Dangerfile`s that could be imported by the projects (e.g. possibly have a `woo/Dangerfile` inside `dangermattic`), but I haven't reached the point yet to decide to go for this approach instead of using more specific plugins.
For now, I just ported what was in Peril to then decide how to move the code around once we start integrating these checks into the projects.

Reply:
> I think this is what is the key here: context of a Danger plugin vs. more specific checks.

🤔 it's tough to know what to do a priori and I don't want my questions/suggestions to get us bogged down implementing abstractions that make sense but that we don't need to use yet.

Given "In the context of a Danger plugin, I fell like these would be useful defaults only for WooCommerce," I guess as long as the plugin/check is only used in Woo the setup does the job. We can iterated later when using it on another app.